src/time_utils.c

```diff
@@ -41,6 +41,45 @@ subtract_interval_from_now(Oid timetype, const Interval *interval)
 	return res;
 }
 
+Datum
+ts_time_datum_convert_arg(Datum arg, Oid *argtype, Oid timetype)
+{
+	Oid type = *argtype;
+
+	if (!OidIsValid(type) || type == UNKNOWNOID)
+	{
+		Oid infuncid = InvalidOid;
+		Oid typeioparam;
+
+		type = timetype;
+		getTypeInputInfo(type, &infuncid, &typeioparam);
+
+		switch (get_func_nargs(infuncid))
+		{
+			case 1:
+				/* Functions that take one input argument, e.g., the Date function */
+				arg = OidFunctionCall1(infuncid, arg);
+				break;
+			case 3:
+				/* Timestamp functions take three input arguments */
+				arg = OidFunctionCall3(infuncid,
+									   arg,
+									   ObjectIdGetDatum(InvalidOid),
+									   Int32GetDatum(-1));
+				break;
+			default:
+				ereport(ERROR,
+						(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+						 errmsg("invalid time argument"),
+						 errhint("Time argument requires an explicit cast.")));
```

Comment:
Is it possible to test this case? Codecov complains that it's not covered.

Reply:
I think most of those tested are already covered by continuous agg refresh function tests, which are reusing the same functionality.

Reply:
I guess Codecov would not complain if it is covered. I searched that such error is never expected in our tests.

Reply:
I don't think this case is actually possible right now, it was made more like a precaution in case new time functions will arrive in future versions I believe